st2common/st2common/models/db/stormbase.py

```diff
@@ -55,10 +55,9 @@ class StormFoundationDB(me.Document, DictSerializableClassMixin):
     # We explicitly assign the manager so pylint know what type objects is
     objects = me.queryset.QuerySetManager()
 
-    # ObjectIdField should be not have any constraints like required,
-    # unique etc for it to be auto-generated.
-    # TODO: Work out how we can mark this as a unique primary key.
-    id = me.ObjectIdField()
+    # Note: In mongoengine >= 0.10 "id" field is automatically declared on all
```

Comment:
I tried to get it work by having `id` explicitly declared (that's my preferred approach) and without needing to have pylint plugin (explicit is better than implicit), but I couldn't get it to work without patching mongoengine - it breaks tons of things and part of it seems to be related to https://github.com/MongoEngine/mongoengine/pull/961.